src/lib/common/selector.py

```diff
@@ -51,18 +61,20 @@ def index(self, config):
         NOTE: should be a relatively light pass that designates the space to be retrieved.
         No options for parallelisation, run on a single CPU.
         """
-        return NotImplemented
+        raise NotImplementedError
+        return pd.DataFrame()
```

Comment:
do we still need this return statement if the `NotImplementedError` is always being raised?

Reply:
removed